miden-lib/asm/miden/tx.masm

```diff
@@ -58,19 +58,36 @@ end
 
 #! Creates a new note and returns a pointer to the memory address at which the note is stored.
 #!
-#! Inputs: [ASSET, tag, note_type, RECIPIENT]
-#! Outputs: [ptr]
+#! Inputs: [tag, aux, note_type, RECIPIENT]
+#! Outputs: [ptr, ZERO, 0, 0]
 #!
-#! ASSET is the asset to be included in the note.
 #! tag is the tag to be included in the note.
-#! note_type is the storage type of the note
+#! aux is an auxiliary field that can be used by the user.
+#! note_type is the storage type of the note.
 #! RECIPIENT is the recipient of the note.
 #! ptr is the pointer to the memory address at which the note is stored.
 export.create_note
     syscall.create_note
-    # => [ptr, ZERO, ZERO, 0]
+    # => [ptr, ZERO, 0, 0]
 
     # clear the padding from the kernel response
-    movdn.8 dropw dropw swap drop
+    movdn.6 dropw drop drop
+    # => [ptr]
+end
+
+#! Adds an ASSET to the specified note and returns a pointer to the memory address 
+#! at which the note is stored.
+#!
+#! Inputs: [ptr, ASSET]
+#! Outputs: [ptr, ZERO]
+#!
+#! ptr is the pointer to the memory address at which the note is stored.
+#! ASSET can be a fungible or non-fungible asset.
+export.add_asset_to_note
```

Comment:
@bobbinth @hackaugusto actually here, we want to use 

`exec.account::remove_asset` to remove the asset before we put it on the note. 

Right now, this is blocked by our `MockDataStore` which is the basis for all our tests. 

We need to refactor that first, before we can make `create_note` cleaner.

Reply:
This is a question for @bobbinth . I don't know why `create_note` doesn't call `remove_asset`. It seems the user is expected to manually handle these details.